yarn-project/aztec-rpc/src/kernel_prover/proof_creator.ts

```diff
@@ -109,33 +106,40 @@ export class KernelProofCreator implements ProofCreator {
     return publicInputs.newCommitments.map(commitment => siloCommitment(wasm, contractAddress, commitment));
   }
 
-  public async createProofInit(txRequest: TxRequest, privateCallData: PrivateCallData): Promise<ProofOutput> {
+  public async createProofInit(privateInputs: PrivateKernelInputsInit): Promise<ProofOutput> {
     const wasm = await CircuitsWasm.get();
     this.log('Executing private kernel simulation init...');
-    const publicInputs = privateKernelSimInit(wasm, txRequest, privateCallData);
+    const result = privateKernelSimInit(wasm, privateInputs);
+    if (result instanceof CircuitError) {
+      throw new CircuitError(result.code, result.message);
+    }
     this.log('Skipping private kernel init proving...');
     // TODO
     const proof = makeEmptyProof();
     this.log('Kernel Prover Init Completed!');
 
+    const publicInputs = result as KernelCircuitPublicInputs;
```

Comment:
Hmm I would think the case wouldn't be needed if we're refining against ! instanceof CircuitError (via the if statement)

Reply:
Good point. To make it work I needed to annotate explicitly the member name of each field in the returned anonymous object.